src/controller/CHIPDeviceController.cpp

```diff
@@ -1083,13 +1092,30 @@ void DeviceCommissioner::OnFailedToExtendedArmFailSafeFailedDeviceAttestation(vo
     commissioner->CommissioningStageComplete(CHIP_ERROR_INTERNAL, report);
 }
 
-void DeviceCommissioner::ExtendArmFailSafeForFailedDeviceAttestation(AttestationVerificationResult result)
+void DeviceCommissioner::ClearAttestationDeviceInfo()
+{
+    if (mAttestationDeviceInfo != nullptr)
+    {
+        chip::Platform::Delete(mAttestationDeviceInfo);
+        mAttestationDeviceInfo = nullptr;
+    }
+}
+
+void DeviceCommissioner::ExtendArmFailSafeForDeviceAttestation(const Credentials::DeviceAttestationVerifier::AttestationInfo & info,
+                                                               Credentials::AttestationVerificationResult result)
 {
     mAttestationResult = result;
 
     auto & params                                                      = mDefaultCommissioner->GetCommissioningParameters();
     Credentials::DeviceAttestationDelegate * deviceAttestationDelegate = params.GetDeviceAttestationDelegate();
-    auto expiryLengthSeconds                                           = deviceAttestationDelegate->FailSafeExpiryTimeoutSecs();
+
+    if (deviceAttestationDelegate->ShouldWaitAfterDeviceAttestation())
+    {
+        ClearAttestationDeviceInfo();
+        mAttestationDeviceInfo = chip::Platform::New<Credentials::DeviceAttestationVerifier::AttestationDeviceInfo>(info);
```

Comment:
Might make sense to make `mAttestationDeviceInfo` a `Platform::UniquePtr` and use `Platform::MakeUnique` here...